### PR TITLE
Update go and distroless-iptables

### PR DESCRIPTION
**Pull request by cpanato**
- bump go-version file to 1.20.4
- bump distroless-iptables to v0.2.3

/assign @aojea

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/kind
 
-go 1.14
+go 1.20
```
Comment (cpanato) [inline go.mod:3]: Any specific reason that was still in 1.14?

Comment (BenTheElder) [reply go.mod:3]: Yes
/hold
See: https://github.com/kubernetes-sigs/kind/pull/3157

Note that go.mod go version implies the go feature level, NOT the go version used for building. 

That would be .go-version here in most cases.

File: .go-version [hunk 2]
```diff
@@ -1 +1 @@
-1.20.2
+1.20.3
```
Comment (BenTheElder) [inline .go-version:1]: This should be 1.20.4 today.

Comment (cpanato) [reply .go-version:1]: they were faster :)

File: go.mod (hunk 1 above)
Comment (cpanato) [reply go.mod:3]: @BenTheElder i will drop this change for now to not block this PR, is that ok?

Comment (BenTheElder) [reply go.mod]: Yes, please revert changes to the go.mod, we're probably going to bump to 1.16 soon but we're not using any go 1.20 features and this is risky for library consumers

**Comment by BenTheElder** [comment]
thanks!

**Pull request merged**